components/autocomplete/index.js

```diff
@@ -176,40 +351,76 @@ class Autocomplete extends Component {
 		// Any handled keycode should prevent original behavior. This relies on
 		// the early return in the default case.
 		event.preventDefault();
-		event.stopImmediatePropagation();
+		event.stopPropagation();
+	}
+
+	toggleKeyEvents( isListening ) {
+		// This exists because we must capture ENTER key presses before Editable.
+		// It seems that react fires the simulated capturing events after the
+		// native browser event has already bubbled so we can't stopPropagation
+		//  and avoid Editable getting the event from TinyMCE, hence we must
+		// register a native event handler.
+		if ( ! this.node ) {
+			return;
+		}
+		const realNode = findDOMNode( this.node );
+		const handler = isListening ? 'addEventListener' : 'removeEventListener';
+		realNode[ handler ]( 'keydown', this.setSelectedIndex, true );
+	}
+
+	componentWillMount() {
+		const { completers } = this.props;
+		// Some completers must do asynchronous requests to get their options so
+		// we calculate that here.
+		completers.forEach( ( { getOptions }, idx ) => {
+			getOptions().then( ( options ) => {
```

Comment:
This behavior means we're issuing one request to the users endpoint for every paragraph block in a post. I'm not sure we need to be so proactive about querying this either? In some cases won't we need fresh data in response to a user search?

Reply:
Good point - I will change it to populate the cache on the first non-empty search of a lookup.

Reply:
Now it loads options on first use.

Reply:
I have tweaked it so it also makes the getOptions call each time the autocompleter is initially triggered, so the information will be fresh, though query characters that narrow down the search will not cause a new getOptions call.